src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java

```diff
@@ -72,6 +72,28 @@ public boolean start(ClickHouseSinkConfig csc) {
             return false;
         }
 
+        try {
+            String chVersion = chc.version();
+            LOGGER.info("Connected to ClickHouse version: {}", chVersion);
+            String[] versionParts = chVersion.split("\\.");
+            if (versionParts.length < 2) {
+                LOGGER.error("Unable to determine ClickHouse server version.");
+                return false;
+            }
+
+            int majorVersion = Integer.parseInt(versionParts[0]);
+            int minorVersion = Integer.parseInt(versionParts[1]);
+            if (majorVersion < 22 || (majorVersion == 22 && minorVersion < 5)) {
```

Comment:
What is the earliest version we test with? I'd expect 23.3, based on https://github.com/ClickHouse/ClickHouse/blob/master/SECURITY.md

Reply:
There is java implementation in `clickhouse-java` related to versions https://github.com/ClickHouse/clickhouse-java/blob/main/clickhouse-data/src/main/java/com/clickhouse/data/ClickHouseVersion.java

Why do we need a version test?